projects/clas12/targets/materials.py

```diff
@@ -0,0 +1,226 @@
+	gmaterial.description = 'carbon fiber material is epoxy and carbon - 1.75g/cm3'
+	gmaterial.density = 1.75
+	gmaterial.addMaterialWithFractionalMass('G4_C', 0.745)
+	gmaterial.addMaterialWithFractionalMass('epoxy', 0.255)
+	gmaterial.publish(configuration)
+
+	# HDIce
+	H_atomic_weight = 1.00784
+	D_atomic_weight = 2.014
+	H_mass_fraction = H_atomic_weight/(H_atomic_weight+D_atomic_weight)
+	D_mass_fraction = D_atomic_weight/(H_atomic_weight+D_atomic_weight)
+	gmaterial = GMaterial('HDIce')
+	gmaterial.description = 'solid HD ice'
+	gmaterial.density = 0.147
+	gmaterial.addMaterialWithFractionalMass('G4_H', H_mass_fraction)
+	gmaterial.addMaterialWithFractionalMass('DeuteriumGas', D_mass_fraction)
+	gmaterial.publish(configuration)
+
+	#HDIce+Al
+	HD_mass_fraction = 1 - (0.175-0.147)/(2.7-0.147)
+	Al_mass_fraction = (0.175-0.147)/(2.7-0.147)
+	gmaterial = GMaterial('solidHD')
+	gmaterial.description = 'solidHD target material - HDIce + Al'
+	gmaterial.density =0.175
+	gmaterial.addMaterialWithFractionalMass('HDIce', HD_mass_fraction)
+	gmaterial.addMaterialWithFractionalMass('G4_Al', Al_mass_fraction)
+	gmaterial.publish(configuration)
+
+	#MgB2
+	Mg_atomic_weight = 24.305
+	B_atomic_weight = 10.811
+	Mg_mass_fraction = Mg_atomic_weight/(Mg_atomic_weight+2*B_atomic_weight)
+	B_mass_fraction = 2*B_atomic_weight/(Mg_atomic_weight+2*B_atomic_weight)
+	gmaterial = GMaterial('MgB2')
+	gmaterial.description = 'MgB2'
+	gmaterial.density = 2.57
+	gmaterial.addMaterialWithFractionalMass('G4_Mg', Mg_mass_fraction)
+	gmaterial.addMaterialWithFractionalMass('G4_B', B_mass_fraction)
+	gmaterial.publish(configuration)
+
+	#CTFE = C_2ClF_3
+	my_density = 2.135 # 2 C, 3 F, 1 Cl
+	C_mass_fraction=2*12/(2*12+3*19+35)
+	F_mass_fraction=3*19/(2*12+3*19+35)
+	Cl_mass_fraction=35/(2*12+3*19+35)
+	gmaterial = GMaterial('Kel-F')
+	gmaterial.description = 'Kel-F PCTFE target walls C_2ClF_3'
+	gmaterial.density = my_density
+	gmaterial.addMaterialWithFractionalMass('G4_C', C_mass_fraction)
+	gmaterial.addMaterialWithFractionalMass('G4_Cl', Cl_mass_fraction)
+	gmaterial.addMaterialWithFractionalMass('G4_F', F_mass_fraction)
+	gmaterial.publish(configuration)
+
+	#Alloy of Cu-Ni
+	my_density = 8.95
+	Cu_mass_fraction=0.7
+	Ni_mass_fraction=0.3
+	gmaterial = GMaterial('Cu70Ni30')
+	gmaterial.description = 'Cupronickel 70-30'
+	gmaterial.density = my_density
+	gmaterial.addMaterialWithFractionalMass('G4_Cu', Cu_mass_fraction)
+	gmaterial.addMaterialWithFractionalMass('G4_Ni', Ni_mass_fraction)
+	gmaterial.publish(configuration)	
+
+	#Polarized He3 (longitudinal target)
+	gmaterial = GMaterial('polarizedHe3')
+	gmaterial.description = 'polarizedHe3 target material'
+	gmaterial.density = 0.000748
+	gmaterial.addMaterialWithFractionalMass('Helium3Gas', 1)
+	gmaterial.publish(configuration)
+
+	# lHe coolant
+	gmaterial = GMaterial('lHeCoolant')
+	gmaterial.description = 'liquid He coolant for the polarized target cell'
+	# @mariakzurek: Should it be 145 or 147??????
```

Comment:
@maureeungaro Please check this comment

Reply:
Good question I do not know! This was written by the transverse group. Small difference though, I'd just pick one: 146!